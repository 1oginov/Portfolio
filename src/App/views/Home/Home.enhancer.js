/* @flow */

import { compose, type HOC } from 'recompose';

import withPositionsArray from 'Portfolio/enhancers/withPositionsArray';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import * as R from '../../routes';

const enhancer: HOC<*, *> = compose(
  withNavigationHandlers({
    handleCreatePositionClick: R.CREATE_POSITION,
    handlePositionClick: (props, position) => ({
      params: { position: position.id },
      route: R.POSITION,
    }),
    handleProfileClick: R.PROFILE,
  }),
  withPositionsArray,
);

export default enhancer;
