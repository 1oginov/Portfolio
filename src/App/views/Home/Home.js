/* @flow */

import * as React from 'react';

const Home = ({ handleCreatePositionClick, handleProfileClick }) => (
  <React.Fragment>

    <h1>Home</h1>

    <div>
      <button onClick={handleCreatePositionClick}>Create position</button>
      <button onClick={handleProfileClick}>Profile</button>
    </div>

  </React.Fragment>
);

export default Home;
