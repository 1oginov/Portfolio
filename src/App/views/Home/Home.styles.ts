import { withStyles } from '@material-ui/core';

import { viewHeadline, viewRoot } from 'Styles/mixins';

export default withStyles(theme => ({
  bar: {
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
  headline: viewHeadline(theme),
  root: {
    ...viewRoot(theme),
    paddingBottom: theme.spacing(2),
    paddingLeft: 0,
    paddingRight: 0,
    paddingTop: theme.spacing(2),
  },
}));
