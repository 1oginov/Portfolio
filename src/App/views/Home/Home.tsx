import * as React from 'react';

import { Position } from 'Portfolio/lib';
import Money from 'Shared/components/Money';
import Progress from 'Shared/components/Progress';

import OpenPositionsList from '../../components/OpenPositionsList';
import StockPositionsValue, {
  RenderProps as StockPositionsValueRenderProps,
} from '../../components/StockPositionsValue';

export interface Props {
  handlePositionClick: (position: Position) => void;
  positions: Position[];
  positionsLoading: boolean;
}

const Home: React.FunctionComponent<Props> = ({
  handlePositionClick, positions, positionsLoading,
}: Props) => (
  <React.Fragment>

    <h1>Home</h1>

    <div>
      {positionsLoading
        ? <Progress />
        : (
          <StockPositionsValue positions={positions}>
            {({ value }: StockPositionsValueRenderProps) => <Money value={value} />}
          </StockPositionsValue>
        )}
    </div>

    <div>
      {positionsLoading
        ? <Progress />
        : <OpenPositionsList onClick={handlePositionClick} positions={positions} />}
    </div>

  </React.Fragment>
);

export default Home;
