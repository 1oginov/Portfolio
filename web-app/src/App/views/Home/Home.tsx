import { Tab, Tabs } from '@material-ui/core';
import * as React from 'react';

import { Progress } from 'Layers/Presentation/Components/Progress';

import ClosedPositionsList from '../../components/ClosedPositionsList';
import OpenPositionsList from '../../components/OpenPositionsList';
import OpenPositionsSummariesList from '../../components/OpenPositionsSummariesList';
import Totals from '../../components/Totals';
import { StockPosition } from '../../lib';

export interface Props {
  classes: { [key: string]: string };
  handlePositionClick: (positionId: string) => void;
  handleTabChange: (tab: 'closed' | 'open' | 'summary') => void;
  positionsLoading: boolean;
  stockPositions: StockPosition[];
  tab: 'closed' | 'open' | 'summary';
}

const Home: React.FunctionComponent<Props> = ({
  classes, handlePositionClick, handleTabChange, positionsLoading, stockPositions, tab,
}: Props) => (
  <>

    {!positionsLoading && (
      <div className={classes.bar}>

        <Totals showClosed={tab === 'closed'} stockPositions={stockPositions} />

        <Tabs
          indicatorColor="primary"
          onChange={(event, newTab) => handleTabChange(newTab)}
          value={tab}
          variant="fullWidth"
        >
          <Tab label="Summary" value="summary" />
          <Tab label="Open" value="open" />
          <Tab label="Closed" value="closed" />
        </Tabs>

      </div>
    )}

    <div className={classes.root}>
      {positionsLoading ? <Progress /> : (
        <>
          {tab === 'closed' && (
            <ClosedPositionsList onPositionClick={handlePositionClick} stockPositions={stockPositions} />
          )}
          {tab === 'open' && (
            <OpenPositionsList onPositionClick={handlePositionClick} stockPositions={stockPositions} />
          )}
          {tab === 'summary' && (
            <OpenPositionsSummariesList stockPositions={stockPositions} />
          )}
        </>
      )}
    </div>

  </>
);

export default Home;
