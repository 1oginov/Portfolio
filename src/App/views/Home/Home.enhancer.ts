import { compose } from 'recompose';

import withPositions from 'Portfolio/enhancers/withPositions';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import { Props } from './Home';
import * as R from '../../routes';

export default compose<Props, {}>(
  withNavigationHandlers({
    handlePositionClick: (props, position) => ({
      params: { position: position.id },
      route: R.POSITION,
    }),
  }),
  withPositions,
);
