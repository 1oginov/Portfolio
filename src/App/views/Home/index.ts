import Home from './Home';
import enhance from './Home.enhancer';
import style from './Home.styles';

export default style(enhance(Home));
