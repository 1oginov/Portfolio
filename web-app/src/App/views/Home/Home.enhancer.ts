import { connect } from 'react-redux';
import { RouteComponentProps as WithRouterProps, withRouter } from 'react-router-dom';
import { compose, withHandlers } from 'recompose';

import { PositionsConnector } from 'Layers/Adapter/Connectors/PositionsConnector/PositionsConnector';
import {
  PositionsFirestoreConnector,
} from 'Layers/Adapter/FirestoreConnectors/PositionsFirestoreConnector/PositionsFirestoreConnector';
import { changeHomeTab } from 'Layers/Application/ActionCreators/HomeActionCreators/HomeActionCreators';
import State from 'State';

import withStockPositions from '../../enhancers/withStockPositions';
import { Props } from './Home';
import * as R from '../../routes';

interface StateProps {
  tab: 'closed' | 'open' | 'summary';
}

interface WithHandlersProps {
  handlePositionClick: (positionId: string) => void;
}

const mapStateToProps = ({ home: { homeTab } }: State): StateProps => ({ tab: homeTab });

const mapDispatchToProps = { handleTabChange: changeHomeTab };

export default compose<Props, Record<string, never>>(
  withRouter,
  withHandlers<WithRouterProps, WithHandlersProps>({

    handlePositionClick: ({ history }) => (positionId: string) => {
      history.push(R.toPosition(positionId));
    },

  }),
  connect(mapStateToProps, mapDispatchToProps),
  PositionsFirestoreConnector,
  PositionsConnector,
  withStockPositions,
);
