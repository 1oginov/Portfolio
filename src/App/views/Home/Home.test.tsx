import { shallow } from 'enzyme';
import * as React from 'react';

import Home from './Home';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Home handlePositionClick={() => undefined} positions={[]} positionsLoading={false} />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when positions are loading', () => {
  const wrapper = shallow(
    <Home handlePositionClick={() => undefined} positions={[]} positionsLoading />,
  );

  expect(wrapper).toMatchSnapshot();
});
