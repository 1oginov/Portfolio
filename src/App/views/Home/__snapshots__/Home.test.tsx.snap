// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Home
  </h1>
  <div>
    <withProps(Connect(lifecycle(mapProps(withProps(StockPositionsValue)))))
      positions={Array []}
    >
      <Component />
    </withProps(Connect(lifecycle(mapProps(withProps(StockPositionsValue)))))>
  </div>
  <div>
    <OpenPositionsList
      onClick={[Function]}
      positions={Array []}
    />
  </div>
</Fragment>
`;

exports[`matches snapshot when positions are loading 1`] = `
<Fragment>
  <h1>
    Home
  </h1>
  <div>
    <Progress />
  </div>
  <div>
    <Progress />
  </div>
</Fragment>
`;
